src/core/score_map.cc

```diff
@@ -142,4 +142,50 @@ detail::SdsScorePair ScoreMap::iterator::BreakToPair(void* obj) {
   return detail::SdsScorePair(f, GetValue(f));
 }
 
+bool ScoreMap::iterator::ReallocIfNeeded(float ratio) {
+  // Unwrap all links to correctly call SetObject()
+  auto* ptr = curr_entry_;
+  while (ptr->IsLink())
+    ptr = ptr->AsLink();
+
+  auto* obj = ptr->GetObject();
+  auto [new_obj, realloced] = static_cast<ScoreMap*>(owner_)->ReallocIfNeeded(obj, ratio);
+  ptr->SetObject(new_obj);
+
+  return realloced;
+}
+
+pair<sds, bool> ScoreMap::ReallocIfNeeded(void* obj, float ratio) {
```

Comment:
I can extra this to seperate function. It's almost identical as to `StringMap`... I will follow up on this on separate PR with a small refactor

Reply:
i think it does not even access `this`. can be anonymous in this file (not part of this class).